src/Firebase/AppCheck/DecodedAppCheckToken.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Kreait\Firebase\AppCheck;
+
+use JsonSerializable;
+
+/**
+ * @phpstan-type DecodedAppCheckTokenShape array{
+ *     app_id: string,
+ *     aud: array<string>,
+ *     exp: string,
+ *     iat: string,
+ *     iss: string,
+ *     sub: string,
+ * }
+ */
+final class DecodedAppCheckToken implements JsonSerializable
+{
+    /**
+     * @var non-empty-string
+     *
+     * @param array<string> $aud
+     */
+    private function __construct(
+        private string $app_id,
+        private array $aud,
+        private string $exp,
+        private string $iat,
+        private string $iss,
+        private string $sub,
+    ) {
+    }
```

Comment:
```suggestion
    private function __construct(
        public readonly string $app_id,
        /** @var array<string> $aud */
        public readonly array $aud,
        public readonly string $exp,
        public readonly string $iat,
        public readonly string $iss,
        public readonly string $sub,
    ) {
    }
```

Like the others above, this would help us get rid of all the related getter methods.